### PR TITLE
chore: Add tests for custom (de-)serializers

### PR DESCRIPTION
**Pull request by marikaner**
This adds unit tests to ensure that custom (de-)serializers are treated correctly.

If you see missing cases, please let me know in the review.

Related to SAP/cloud-sdk-backlog#399.

### CODE REVIEW

File: packages/odata-common/src/entity-base.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export interface EntityApi<
   requestBuilder(): RequestBuilder<EntityT, DeSerializersT>;
   entityBuilder(): EntityBuilderType<EntityT, DeSerializersT>;
   entityConstructor: Constructable<EntityT>;
-  schema: Record<string, any>;
+  schema;
```
Comment (FrankEssenberger) [inline packages/odata-common/src/entity-base.ts:40]: no type on base? Overwritten on generate entites?

File: packages/odata-common/src/entity-deserializer.ts [hunk 2]
```diff
@@ -79,7 +79,7 @@ export function entityDeserializer<T extends DeSerializers>(
     requestHeader?: any
   ): EntityT {
     const etag = extractODataETag(json) || extractEtagFromHeader(requestHeader);
-    return Object.values(entityApi.schema)
+    return Object.values(entityApi.schema as Record<string, any>)
```
Comment (FrankEssenberger) [inline packages/odata-common/src/entity-deserializer.ts:82]: here you enforce the type you removed in entityBase - can it stay in entityBase?

File: test-resources/test/test-util/custom-de-serializers.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const customStringPropertyValue = 100;
+
+export const customTestDeSerializers = {
```
Comment (FrankEssenberger) [inline test-resources/test/test-util/custom-de-serializers.ts:3]: can you re-use this in the type-tests where I defined a custom deserializer a few time. In batch type test for sure but also in de-serializer type tests

File: packages/odata-v2/src/uri-conversion/odata-uri.spec.ts [hunk 4]
```diff
@@ -64,7 +71,7 @@ describe('getFilter', () => {
     );
   });
 
-  it('for nested filters with and operator only', () => {
+  it("for nested filters with 'and' operator only", () => {
```
Comment (FrankEssenberger) [inline packages/odata-v2/src/uri-conversion/odata-uri.spec.ts:74]: tI know it was the way before, but for me it is not clear what it does. I prefer a verb like 'transforms' or so.

File: packages/odata-v2/src/uri-conversion/odata-uri.spec.ts [hunk 5]
```diff
@@ -303,6 +310,30 @@ describe('getFilter for filter functions', () => {
   });
 });
 
+describe('getFilter for custom (de-)serializers', () => {
+  const oDataUriCustom = createODataUri(
+    mergeDefaultDeSerializersWith(customTestDeSerializers)
+  );
+
+  it('for simple filter', () => {
```
Comment (FrankEssenberger) [inline packages/odata-v2/src/uri-conversion/odata-uri.spec.ts:318]: minor: also here I prefer it which reads more like a sentence.

File: packages/odata-v4/src/uri-conversion/odata-uri.spec.ts [hunk 6]
```diff
@@ -106,3 +111,16 @@ describe('getFilter', () => {
     ).toBe(encodeURIComponent('fn() eq [1,2,3]'));
   });
 });
+
+describe('getFilter for custom (de-)serializers', () => {
+  const { getFilter } = createODataUri(
+    mergeDefaultDeSerializersWith(customTestDeSerializers)
+  );
+
+  it('for lambda expression', () => {
```
Comment (FrankEssenberger) [inline packages/odata-v4/src/uri-conversion/odata-uri.spec.ts:120]: see above

**Review by FrankEssenberger — APPROVED** [review]
LGTM - only very minor comments or questions

File: packages/odata-common/src/entity-deserializer.ts (hunk 2 above)
Comment (marikaner) [reply packages/odata-common/src/entity-deserializer.ts:82]: The idea was to improve the types on `testEntityApi.schema.TO_SINGLE_LINK._linkedEntityApi.schema...` but it didn't work ;) This should be fixed on your PR.

File: packages/odata-common/src/entity-base.ts (hunk 1 above)
Comment (marikaner) [reply packages/odata-common/src/entity-base.ts]: Done.

File: test-resources/test/test-util/custom-de-serializers.ts (hunk 3 above)
Comment (marikaner) [reply test-resources/test/test-util/custom-de-serializers.ts:3]: done

File: packages/odata-v2/src/uri-conversion/odata-uri.spec.ts (hunk 4 above)
Comment (marikaner) [reply packages/odata-v2/src/uri-conversion/odata-uri.spec.ts]: I agree, done.

**Pull request merged**